### ISSUE TITLE
Problem with URL multilanguage

### ISSUE THREAD

**Issue by canon4o** [cakephp/cakephp#7556]
Hi guys!

I have problem with cyrillic URL. Maybe the problem is in my code, but my solve is in the Router class.
https://github.com/cakephp/cakephp/blob/master/src/Routing/Router.php#L338

When I parse url, it should be call urldecode() function.Thus routes can work with cyrillic urls.

**Comment by markstory** [comment]
Do you decode before parse() is called? What is a sample URL that is giving you trouble?

**Comment by canon4o** [comment]
Uhm, no. 
It is about for routes. My sample is:

I have in routes.php file:
```php
Router::defaultRouteClass('DashedRoute');
    Router::scope('/', function ($routes) {
        $routes->redirect('/', '/ден/15-октомври');

        $routes->connect('/ден/:day-:month', [
            'controller' => 'Events',
            'action' => 'day'
        ], [
            'day' => '0[1-9]|[12][0-9]|3[01]',
            'month' => implode('|', ['януари', 'февруари', 'март', 'април', 'май', 'юни', 'юли', 'август', 'септември', 'октомври', 'ноември', 'декември']),
            'pass' => ['day', 'month']
        ]);
    });
    Plugin::routes();
```
And when I open the site, I get this error.
> Error: A route matching "/%D0%B4%D0%B5%D0%BD/17-%D0%BE%D0%BA%D1%82%D0%BE%D0%BC%D0%B2%D1%80%D0%B8" could not be found.

**Comment by markstory** [comment]
Ok, what your do you visit in your browser to trigger the issue with those routes?

**Comment by canon4o** [comment]
http://localhost or http://localhost/ден/15-октомври

**Comment by markstory** [comment]
Thanks, I'll see if I can reproduce locally.

**Comment by markstory** [comment]
Fixed with #7590
